dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/AclEventExporterTest.java

```diff
@@ -423,6 +426,41 @@ void shouldReturnNoEventsGivenUserHasNoAccess() throws ForbiddenException, BadRe
     assertIsEmpty(events);
   }
 
+  @Test
+  void shouldReturnAllOrgUnitEventsWhenOrgUnitModeAllAndNoOrgUnitProvided()
+      throws ForbiddenException, BadRequestException {
+    injectSecurityContext(userService.getUser("lPaILkLkgOM"));
+
+    EventOperationParams params = EventOperationParams.builder().orgUnitMode(ALL).build();
+
+    List<Event> events = eventService.getEvents(params).getEvents();
+
+    assertFalse(
+        events.isEmpty(),
+        "Expected to find events when ou mode ALL no program specified and no org unit provided");
+    assertContainsOnly(
+        List.of("h4w96yEMlzO", "uoNW0E3xXUy", "DiszpKrYNg8", "tSsGrtfRzjY"),
+        events.stream().map(e -> e.getOrganisationUnit().getUid()).collect(Collectors.toSet()));
+  }
+
+  @Test
+  void shouldReturnAllOrgUnitEventsWhenOrgUnitModeAllAndOrgUnitProvided()
+      throws ForbiddenException, BadRequestException {
+    injectSecurityContext(userService.getUser("lPaILkLkgOM"));
+
+    EventOperationParams params =
+        EventOperationParams.builder().orgUnitUid("uoNW0E3xXUy").orgUnitMode(ALL).build();
+
+    List<Event> events = eventService.getEvents(params).getEvents();
+
+    assertFalse(
+        events.isEmpty(),
+        "Expected to find events when ou mode ALL no program specified and no org unit provided");
```

Comment:
```suggestion
        "Expected to find events when ou mode ALL no program specified and org unit provided");
```

Does it make sense to accept mode ALL with an orgUnit? Because in a RequestParamValidator you reject a request like that with `orgUnitMode %s cannot be used with orgUnits. Please remove the orgUnit parameter and try again.`

Or is it due to the layers and that this layer is testing via the service (operation params)? So the web layer adds the orgUnit to the params even for mode ALL. What orgUnit will that be then?

Reply:
You are right that the API will reject a request containing an org unit if the mode is ALL.

However, it is my understanding that we have jobs running as admin calling the service directly, thus bypassing the the web layer validations. 
For that reason I added a test to validate that even if an org unit is used with mode ALL, this org unit will be ignored when retrieving the results from the database.

I changed the test name to `shouldIgnoreRequestedOrgUnitAndReturnAllEventsWhenOrgUnitModeAllAndOrgUnitProvided`, maybe it's clearer this way.